src/utils/utils.h

```diff
@@ -0,0 +1,70 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef DALI_BACKEND_UTILS_UTILS_H_
+#define DALI_BACKEND_UTILS_UTILS_H_
+
+
+namespace triton { namespace backend { namespace dali {
+
+/**
+ * Splits string according to a delimiter.
+ *
+ * This function doesn't include empty strings in the output.
+ *
+ * @param str String to be split
+ * @param delimiter Delimiter. Won't be included in the output
+ * @return
+ */
+inline std::vector<std::string> split(const std::string& str, const std::string& delimiter = ":") {
+  std::vector<std::string> ret;
+  size_t start = 0, end = 0;
+  do {
+    end = str.find(delimiter, start);
+    auto substring = str.substr(start, end - start);
+    if (!substring.empty()) {
+      ret.emplace_back(substring);
+    }
+    start = end + delimiter.length();
+  } while (end != std::string::npos);
+  return ret;
+}
+
+
+template<typename T>
+T from_string(const std::string& str) {
+  assert(false);  // Not implemented
+}
```

Comment:
If it's not implemented then don't implement it. Why wait with the error until run time?
```suggestion
template<typename T>
T from_string(const std::string& str);
```